### ISSUE TITLE
fork doesn't show untracked stashed files in stash

### ISSUE THREAD

**Issue by weinand** [ForkIssues/Tracker#898]
If I stash changes with the "-u" option to include untracked files,
**fork** does not show the untracked file in the "Changes" view.

Please see https://stackoverflow.com/questions/12681529/in-git-is-there-a-way-to-show-untracked-stashed-files-without-applying-the-stas for details how to show the untracked files.

**Comment by njzydark** [comment]
Also need this feature.

**Comment by brandonzylstra** [comment]
This is the only bug I've encountered in Fork so far.  I'm surprised that despite being reported a long time ago it is still not fixed. 

Here is the `git status` before I stashed:
![image](https://user-images.githubusercontent.com/9854/90820776-ebdeb380-e2ff-11ea-9adf-b9fd82a4fe29.png)

Here is what I see in Fork:
![image](https://user-images.githubusercontent.com/9854/90821064-509a0e00-e300-11ea-90ec-0da705171564.png)
_There's a file missing._

Here is what I see in Tower:
![image](https://user-images.githubusercontent.com/9854/90820901-1892cb00-e300-11ea-852a-cbccc6166018.png)

As far as I know, Tower is the only app that gets this right.  Git Kraken (when I tested) also failed here.  I would love to switch to only using the CLI and Fork.  But for now I'm pulling Tower out whenever I need to view stashes without fear that files are missing.

**Comment by yorch** [comment]
Just realized that fork wasn't showing my untracked files, thought I lost a bunch of work I had stashed. A workaround is to stage the untracked file before creating the stash.

**Comment by brandonzylstra** [comment]
I think Fork does that for you if you use it to stash, but I don't want to mix things together like that.  I want to keep all three types of files separate: added, modified, and new.  Not everything that is new is something I'll want to commit, and when I unstash I want to see things exactly as they were before I stashed.

**Comment by brandonzylstra** [comment]
This is the main reason I have not paid for Fork yet.  💵

**Comment by yorch** [comment]
@brandondrew that's a fair point.. I usually only use Fork when I want to see my stashes.. for the most part, I use [`tig`](https://jonas.github.io/tig/) when I want to quickly check previous commits, and command line + VSCode integrated git support for staging and committing

**Comment by brandonzylstra** [comment]
@yorch Fork is amazing for interactive rebasing.  I do most of my work inside `zsh`, but the interactive rebasing in Fork is better than any shell-based utilities in my opinion, and better than any other GUIs I've seen, and WAAAYYYY better than typing out `git` commands in the shell.  (Lazy Git also looks very nice, and allows nifty things like moving part of a commit to a previous commit, but that particular feature is unfortunately super un-intuitive, and the current version of Lazy Git is totally different from the video its author made explaining this feature, so....)

**Comment by yorch** [comment]
@brandondrew ah, interesting, never tried to the interactive rebase in Fork. I usually use the good old `git` command and `vim`, but yeah, if I have to edit too much stuff in `vim` (as I haven't dominated the dark art of `vim` yet), things get slow.. will definitely give it a try! thanks!

**Comment by grigorye** [comment]
Just in case, we can still see the untracked from a stash, but separately from the rest: we need to navigate to [3rd parent of stash](https://stackoverflow.com/a/12681856/1859783). That "link" is available in "Commit" view of the stash: 
<img width="431" alt="Screen Shot 2022-09-15 at 02 25 45" src="https://user-images.githubusercontent.com/803905/190285681-cd56622b-99f9-472e-81af-db56a31969f8.png">

**Comment by brandonzylstra** [comment]
@grigorye yes, thanks, but that's an extremely inefficient technique for looking through stashes.  For example, imagine I have a stash and I know it has a migration that adds a column to the users table.  I would have to click on every stash and then click on the third parent for each one.  

If Fork did not have this limitation, I could just scan through the stashes with the down arrow until I found one with a migration that was named appropriately.

I love Fork, but this one limitation is the main thing making me reluctant to register it, and the main reason I keep Tower handy.

**Comment by leehericks** [comment]
I used the stash including untracked files option in VSCode. Couldn't see them in Fork. 
Thank god I checked `git stash show -u` myself and found this issue.
Almost cried thinking I lost a lot a lot of code. It's very shocking to look at Fork and think all your work is gone.

**Comment by grigorye** [comment]
Just in case, I entirely agree with the above sentiments: 1) it's really scary when it looks like the changes you stashed are lost 2) hence, there should be a better way to see untracked files in a stash. I can imagine that it's a bit of a challenge to implement the proper support for this (as the untracked part is a separate object from Git perspective). However, it indeed looks like an important thing to fix, provided that the rest of Fork is really high-level/intuitive, compared to this very "scary nuance".

**Comment by seanarnold** [comment]
I just noticed this when looking at a stash from a few weeks ago. It showed 3 files but when applying I got the 10+ others back. It's a very confusing UX and should be fixed imo

**Comment by snowe2010** [comment]
I too thought I'd lost my files and had to recover them with intellij's Local History feature. When I then went to restore a _different_ stash (or so I thought), it wouldn't restore because the files were actually there.

**Comment by ByScripts** [comment]
I love Fork, but could someone explain why this issue (open for more than 6 years) has received absolutely no attention from the Fork team? No triage, no acknowledgment, no "won't fix"... Just silence.

`git stash -u` is a standard and first-class Git command and should be supported by Fork.

Thank you.

**Comment by DanPristupov** [comment]
@ByScripts I gave a very brief answer here: https://github.com/fork-dev/TrackerWin/issues/2624#issuecomment-4721002124. TLDR: "won't fix in the near future".
